go/exoservice/exoservice.go

```diff
@@ -97,17 +97,18 @@ func (e *ExoService) receiveMessage(message structs.Message) {
 	}
 	e.messageHandlers[message.Name](Request{
 		Payload: message.Payload,
-		Reply:   e.buildSendHelper(message.ID),
-		Send:    e.buildSendHelper(""),
+		Reply:   e.buildSendHelper(message.ID, message.SessionID),
+		Send:    e.buildSendHelper(message.ID, message.SessionID),
```

Comment:
I believe the first argument to send should stay as an empty string. Send is meant to respond to a message but have it not be a direct reply.